### PR TITLE
Merge pull request #10892 from anastasiamac/lp1851849-lp1851852

### PR DESCRIPTION
**Pull request by anastasiamac**
https://github.com/juju/juju/pull/10892

## Description of change

add-k8s command is frequently piped. However, the behavior of the command has changed to prompt for user input when neither --client nor --controller option was specified to support an ASK-OR-TELL brief.

Since the pipe and commands that require user input don't work together, if the pipe was used, the command to fail immediately with a descriptive message. 

As a drive-by, rename remaining reference to an ephemeral --client-only option.

## QA steps

1. $ microk8s.config | juju add-k8s k8s
```
This operation can be applied to both a copy on this client and to the one on a controller.
ERROR The command is piped and Juju cannot prompt to clarify whether the --client or a --controller is to be used.
Please clarify by re-running the command with the desired option(s).
```
2. $ juju add-credential aws (to ensure everything else works)
```
This operation can be applied to both a copy on this client and to the one on a controller.
No current controller was detected and there are no registered controllers on this client: either bootstrap one or register one.
Do you ONLY want to add a credential to this client? (Y/n):
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1851849
https://bugs.launchpad.net/juju/+bug/1851852

### CODE REVIEW

**Comment by anastasiamac** [comment]
[deal with --client-only references](https://trello.com/c/31D1l5G0/976-deal-with-client-only-references)

**Comment by anastasiamac** [comment]
[detect when pipe is used and err instead of prompt](https://trello.com/c/JpiJyvw2/977-detect-when-pipe-is-used-and-err-instead-of-prompt)

File: cmd/supercommand.go [hunk 1]
```diff
@@ -86,3 +87,10 @@ func Info(i *cmd.Info) *cmd.Info {
 	info.ShowSuperFlags = []string{"show-log", "debug", "logging-config", "verbose", "quiet", "h", "help"}
 	return &info
 }
+
+// IsPiped determines if the command was used in a pipe and,
+// hence, it's stdin is not usable for user input.
+func IsPiped(ctx *cmd.Context) bool {
```
Comment (timClicks) [inline cmd/supercommand.go]: Do we know if this could produce false positives?

Comment (anastasiamac) [reply cmd/supercommand.go]: I have done the testing on several commands and scenarios and have not encountered any discrepancies... So, if there is such a possibility, we have not come across it yet.

This code is mirror image of what add-k8s command uses to get piped config, and that code is exercised excessively. So if it was unstable, it'd be discovered quickly.

**Comment by anastasiamac** [comment]
$$merge$$

**Comment by anastasiamac** [comment]
14:38:43 grouped write of manifest, lock and vendor: error while writing out vendor tree: failed to write dep tree: failed to export cloud.google.com/go: failed to fetch source for https://code.googlesource.com/gocloud: unable to get repository: Cloning into '/home/ubuntu/go/pkg/dep/sources/https---code.googlesource.com-gocloud'...

**Comment by anastasiamac** [comment]
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
14:38:43 fatal: The remote end hung up unexpectedly
14:38:43 fatal: early EOF
14:38:43 fatal: index-pack failed
14:38:43 : command failed: [git clone --recursive -v --progress https://code.googlesource.com/gocloud /home/ubuntu/go/pkg/dep/sources/https---code.googlesource.com-gocloud]: exit status 128
14:38:43 make: *** [dep] Error 1
14:38:43 Makefile:74: recipe for target 'dep' failed

**Comment by anastasiamac** [comment]
windows check failed with:
14:05:34 2019-11-11 04:02:14 WARNING utils.featureflag flags_windows.go:34 Failed to open juju registry key HKLM:\SOFTWARE\juju-core; feature flags not enabled

**Pull request merged**